src/utils/authorizeDelegatedAccount.ts

```diff
@@ -0,0 +1,52 @@
+import { Contract, Wallet, providers } from 'ethers'
+import * as DelegatedAccessRegistry from '../contracts/DelegatedAccessRegistry.sol/DelegatedAccessRegistry.json'
+
+import { sign, hash } from 'eth-crypto'
+import handleError from '$/utils/handleError'
+import { Address } from '$/types'
+
+const DelegatedAccessRegistryAddress = '0xf5803cdA6352c515Ee11256EAA547BE8422cC4EE'
+
+enum ChallengeType {
+    Authorize = 0,
+    Revoke = 1,
+}
+
+const signDelegatedChallenge = async (
```

Comment:
No `await` used? No `async` needed.

Reply:
TS should be able to infer the type from a `return`.

Reply:
Another thing is that both [`keccak256`](https://github.com/pubkey/eth-crypto/blob/master/src/hash.js#L6-L19) and [`sign`](https://github.com/pubkey/eth-crypto/blob/master/src/sign.js#L16-L30) seem to be sync functions so our `signDelegatedChallenge ` can be too!